web/app/notifications.py

```diff
@@ -117,10 +126,21 @@ def send_failure_alert_pushbullet(printer, is_warning, print_paused):
             pass
 
         if file_url:
-            pb.push_file(file_url=file_url, file_name="Detected Failure.jpg", file_type="image/jpeg", body=body, title=title)
+            pb.push_file(file_url=file_url, file_name='Detected Failure.jpg', file_type='image/jpeg', body=body, title=title)
         else:
             pb.push_link(title, link, body)
     except PushError as e:
         LOGGER.error(e)
     except PushbulletError as e:
         LOGGER.error(e)
+
+def send_failure_alert_telegram(printer, is_warning, print_paused):
+    if not printer.user.telegram_eligible():
+        return
+
+    chat_id = printer.user.telegram_chat_id
+    notification_text = notification_elements(printer, is_warning, print_paused)
+
+    photo = get_photo(printer)
+
+    telegram_bot.send_notification(chat_id, notification_text, photo, printer_id)
```

Comment:
I got `NameError: name 'telegram_bot' is not defined` error on this line when I was testing it. Did it work for you?

Also shouldn't `printer_id` be `printer.id` instead?

Reply:
Ah, I'd mostly tested it through sticking send_notification calls into the printers view for testing, since I haven't hooked my dev copy up to my printer yet. Right now going through and making the suggested changes.